packages/tx/src/transactionFactory.ts

```diff
@@ -0,0 +1,69 @@
+import Common from '@ethereumjs/common'
+import { UnsignedLegacyTransaction, SignedLegacyTransaction } from './unsignedLegacyTransaction'
+import { UnsignedEIP2930Transaction, SignedEIP2930Transaction } from './unsignedEIP2930Transaction'
+import { TxOptions, Transaction } from './types'
+
+const DEFAULT_COMMON = new Common({ chain: 'mainnet' })
+
+export default class TransactionFactory {
+  // It is not possible to instantiate a TransactionFactory object.
+  private constructor() {}
+
+  public static fromRawData(rawData: Buffer, transactionOptions: TxOptions): Transaction {
+    const common = transactionOptions.common ?? DEFAULT_COMMON
+    if (rawData[0] <= 0x7f) {
+      // It is an EIP-2718 Typed Transaction
+      if (!common.eips().includes(2718)) {
+        throw new Error('Cannot create a TypedTransaction: EIP-2718 is not enabled')
+      }
+      // Determine the type.
+      let EIP: number
+      switch (rawData[0]) {
+        case 1:
+          EIP = 2930
+          break
+        default:
+          throw new Error(`TypedTransaction with ID ${rawData[0]} unknown`)
+      }
+
+      if (!common.eips().includes(EIP)) {
+        throw new Error(
+          `Cannot create TypedTransaction with ID ${rawData[0]}: EIP ${EIP} not activated`
+        )
+      }
+
+      return UnsignedEIP2930Transaction.fromRlpSerializedTx(rawData, transactionOptions)
+    } else {
+      return UnsignedLegacyTransaction.fromRlpSerializedTx(rawData, transactionOptions)
+    }
+  }
+
+  /**
+   * This helper method allows one to retrieve the class which matches the transactionID
+   * If transactionID is undefined, return the LegacyTransaction class.
+   * Based on the `signed` boolean, return either the signed version or the unsigned version.
+   * Note that the static methods of the unsigned versions automatically return the signed version,
+   * if that data exists in the parameters.
+   * @param transactionID
+   * @param signed
+   * @param common
+   */
+  public static getTransactionClass(
+    transactionID: number = 0, // Transaction ID 0 is a special type; it is the Legacy Transaction
```

Comment:
Reading from the EIP definitions "[Backwards Compatibility](https://eips.ethereum.org/EIPS/eip-2718#implementation)" section, transaction ID for legacy transactions is rather in a range then being a single number, so just checking for `0` should not be enough I would assume?

Reply:
Ah, I think we are thinking about this in a different way - you are proposing to also include all numbers >= `0x80` here? So you can just check the first byte of the RLP-encoded transaction?

Reply:
I am not proposing anything here, I just thought from a first look that there might be the possibility that you have misread the EIP at that point, but I didn't look into your code too closely. So if you are still confident that the implementation is correct on a second look everything should be ok.

Reply:
Currently anything >= 0x80 and <= 0xff return a LegacyTransaction.

Reply:
lgtm, i see the logic for checking that a few lines down on line 107.